framework/src/iotbus/iotbus_pwm.c

```diff
@@ -192,8 +192,7 @@ int iotbus_pwm_get_duty_cycle(iotbus_pwm_context_h pwm)
 		zdbg("ioctl(PWMIOC_GETCHARACTERISTICS) failed: %d\n", errno);
 		return IOTBUS_ERROR_UNKNOWN;
 	}
-
-	return (int)(info->duty);
+	return (int)((info->duty * 100.0 / 65536) + 0.5);
```

Comment:
why 65536? info->duty resolution is 32 bit it is typedef of uint32_t.
Also do not hard code the resolution. if data type is changed then we will face problem